iocage_lib/ioc_start.py

```diff
@@ -1312,19 +1311,45 @@ def __check_rtsold__(self):
             stdout=su.PIPE
         )
 
-    def get_default_gateway(self):
-        # e.g response - ('192.168.122.1', 'lagg0')
-        try:
-            return netifaces.gateways()["default"][netifaces.AF_INET]
-        except KeyError:
+    def get_default_interface(self):
+        if netifaces.AF_INET in self.host_gateways['default']:
+            return self.host_gateways['default'][netifaces.AF_INET][1]
+        elif netifaces.AF_INET6 in self.host_gateways['default']:
+            return self.host_gateways['default'][netifaces.AF_INET6][1]
+        else:
             iocage_lib.ioc_common.logit(
                 {
                     'level': 'EXCEPTION',
-                    'message': 'No default gateway interface found'
+                    'message': 'No default interface found'
                 },
                 _callback=self.callback,
-                silent=self.silent
-            )
+                silent=self.silent)
+
+    def get_default_ipv4_gateway(self):
+        gateway = ''
+        if netifaces.AF_INET in self.host_gateways['default']:
+            gateway = self.host_gateways['default'][netifaces.AF_INET][0]
+        iocage_lib.ioc_common.logit({
+            'level': 'WARNING',
+            'message': f'{self.uuid}: vnet requires IPv4 defaultrouter,'
+            f' using {gateway}'
+        }, _callback=self.callback,
+            silent=self.silent)
+        return gateway
+
+    def get_default_ipv6_gateway(self):
+        gateway = ''
+        if netifaces.AF_INET6 in self.host_gateways['default']:
+            gateway = re.sub(
+                '%.*$', '%epair0b',
+                self.host_gateways['default'][netifaces.AF_INET6][0])
+        iocage_lib.ioc_common.logit({
+            'level': 'WARNING',
+            'message': f'{self.uuid}: vnet requires IPv6 defaultrouter,'
+            f' using {gateway}'
```

Comment:
This may look a little odd if gateway is actually empty, we should always show them something informative.

Reply:
You are right. This message only made sense in the `if self.conf['vnet'] and self.defaultrouter == 'none'` context:
https://github.com/iocage/iocage/blob/40327cdb6afbbb725e452d2709887e1a221ffa00/iocage_lib/ioc_start.py#L216